src/systems/character-controller-system.js

```diff
@@ -251,14 +261,15 @@ export class CharacterControllerSystem {
 
       newPOV.copy(snapRotatedPOV);
 
+      const navMeshExists = NAV_ZONE in this.scene.systems.nav.pathfinder.zones;
       if (!this.isMotionDisabled) {
         const playerScale = v.setFromMatrixColumn(this.avatarPOV.object3D.matrixWorld, 1).length();
         const triedToMove = this.relativeMotion.lengthSq() > 0.000001;
 
         if (triedToMove) {
           calculateDisplacementToDesiredPOV(
             snapRotatedPOV,
-            this.fly,
+            this.fly || !navMeshExists,
```

Comment:
This makes it so that if there is no nav mesh, the client moves the same way it does when it is in fly mode. However, I chose not to actually enable fly mode (and show a "fly mode enabled" message to the user), so that the user would not have to manually exit fly mode when switching scenes. 

I wonder if we should go even further than this, and disable fly mode toggling when there is no nav mesh. We could show a message like, "cannot toggle fly mode in this scene because there is no floor plan!".